src/renderer/components/wallet/assets/AssetDetails.tsx

```diff
@@ -77,16 +77,19 @@ export const AssetDetails: React.FC<Props> = (props): JSX.Element => {
     )
   }, [oWalletAddress, history])
 
-  const isNonNativeRuneAsset: boolean = useMemo(() => AssetHelper.isNonNativeRuneAsset(asset), [asset])
+  const isNonNativeRuneAsset: boolean = useMemo(
+    () => AssetHelper.isNonNativeRuneAsset(asset, network),
+    [asset, network]
+  )
 
   const walletActionUpgradeNonNativeRuneClick = useCallback(() => {
     FP.pipe(
       oWalletAddress,
       O.filter((_) => isNonNativeRuneAsset),
-      O.map((walletAddress) => walletRoutes.upgradeRune.path({ asset: assetToString(asset), walletAddress })),
+      O.map((walletAddress) => walletRoutes.upgradeRune.path({ asset: assetToString(asset), walletAddress, network })),
```

Comment:
No need to pass `network` to path. All views / components do know about current`network`

Reply:
Remove `network` from `path` ;)